### PR TITLE
feat(start_planner): output collided stop path

### PR DESCRIPTION
**Pull request by danielsanchezaran**
## Description

Sometimes, when the ego vehicle is surrounded by other vehicles/objects, the start planner is unable to find a pull out path that does not collide with other objects and instead outputs a stop path (a path with 2 points and 0 speed). However, if the surrounding vehicles move, or the ego vehicle is moved manually, as in a driver moves the vehicle to a place were it should be able to take off from, the start planner does not output a new path and, instead, the start and goal poses must be set again so the start planner can output a valid path. 

With this PR, if the start planner does not find a non-collided path, it outputs the "best collided path" with a stop point on the ego's current position, and periodically checks if the stop point can be removed (either because the ego was moved manually by a driver or because the surrounding obstacles disappeared), which can be very beneficial when running Autoware on real vehicles. 

Examples: 

1)The start planner outputs a geometric path, but, since it is surrounded by other vehicles a stop point is inserted, once the other vehicles are removed, the ego can move:

https://github.com/autowarefoundation/autoware.universe/assets/25967964/5e6c7966-0b84-46f2-9211-607290408e84

2) The ego outputs a collided path with stop point when it is surrounded by other objects:

-  First, I move the ego position (as if a driver moved the vehicle manually) to a new point in the path where the ego can continue -> the path turns green and the ego can proceed
- Second, same situation, but I eliminate the surrounding vehicles and the ego can proceed.

https://github.com/autowarefoundation/autoware.universe/assets/25967964/d85031d1-7a9e-4efd-99b2-97aeaa8568ea

Limitations: 
Paths with backward movements are not fully supported: 
Since the output paths with backward movement are "divided in several parts" it is difficult at the moment to check in which part of the path the ego vehicle is in order to re-check for collisions with static objects, the result is that, if the driver takes manually and moves the ego to a different position, the "active path" will still be the backward path and the ego wont move.

## Related links

Related Ticket: [TIER IV INTERNAL LINK](https://tier4.atlassian.net/browse/RT1-5261)
Conversation:  [TIER IV INTERNAL LINK](https://star4.slack.com/archives/C03QW0GU6P7/p1701153915688299?thread_ts=1701152695.782779&cid=C03QW0GU6P7)
## Tests performed

PSim
Evaluator: [TIER IVs INTERNAL LINK](https://evaluation.ci.tier4.jp/evaluation/reports/54318455-998c-5918-bb69-ec1bae497c95?project_id=prd_jt)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_start_planner_module/src/util.cpp [hunk 1]
```diff
@@ -72,6 +72,14 @@ PathWithLaneId getBackwardPath(
   return backward_path;
 }
 
+bool backwardsMovementIsNecessary(
```
Comment (kyoichi-sugahara) [inline planning/behavior_path_start_planner_module/src/util.cpp:75]: :heart:

File: planning/behavior_path_start_planner_module/src/start_planner_module.cpp [hunk 2]
```diff
@@ -766,9 +837,9 @@ bool StartPlannerModule::findPullOutPath(
   const Pose & refined_start_pose, const Pose & goal_pose, const double collision_check_margin)
 {
   // if start_pose_candidate is far from refined_start_pose, backward driving is necessary
```
Comment (kyoichi-sugahara) [inline planning/behavior_path_start_planner_module/src/start_planner_module.cpp:839]:
```suggestion

```

**Comment by kyoichi-sugahara** [comment]
highly appreciate this improvement
> periodically checks if the stop point can be removed (either because the ego was moved manually by a driver or because the surrounding obstacles disappeared), which can be very beneficial when running Autoware on real vehicles.

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM!

**Review by kyoichi-sugahara — APPROVED** [review]
nice!

**Comment by danielsanchezaran** [comment]
I am not happy with the current state of this PR and I believe there can be some problems related to chattering, so I am moving it back to a draft. @kyoichi-sugahara

**Comment by kyoichi-sugahara** [comment]
@danielsanchezaran 
Agree! thanks

**Pull request closed without merging**